src/core/detail/btree_internal.h

```diff
@@ -0,0 +1,458 @@
+// Copyright 2023, Roman Gershman.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <cassert>
+#include <cstring>
+
+namespace dfly {
+
+template <typename T, typename Policy> class BTree;
+
+namespace detail {
+
+// Internal classes related to B+tree implementation. The design is largely based on the
```

Comment:
I thought in the opposite direction: we use it only for a small number of types, so it can be explicitly instantiated for those and the internals can be hidden away

Reply:
To not insult anyone, I will leave it as is :)